mmdet/models/backbones/resnet.py

```diff
@@ -473,8 +474,12 @@ def forward(self, x):
 
     def train(self, mode=True):
         super(ResNet, self).train(mode)
-        if mode and self.norm_eval:
-            for m in self.modules():
-                # trick: eval have effect on BatchNorm only
-                if isinstance(m, nn.BatchNorm2d):
-                    m.eval()
+        if mode:
+            self._freeze_stages()
+            if self.norm_eval:
+                for m in self.modules():
+                    # trick: eval have effect on BatchNorm only
+                    if isinstance(m, _BatchNorm):
+                        m.eval()
+        else:
+            convert_batchnorm(self)
```

Comment:
Maybe after `convert_batchnorm`, `eval( )` for BN layers is needed. In the `train` function, `eval( )` is called for whole network in `super(ResNet, self).train(mode)`, but after `convert_batchnorm` function, the BN layers are in `train` mode, which may lead to a significance performance drop.

Reply:
That shouldn't be a problem since the `eval` or `train` is called in recursive way. But I would discard the `convert_batchnorm` since the mmddp testing is available now.